kong/key_service_test.go

```diff
@@ -0,0 +1,544 @@
+	assert.NoError(err)
+}
+
+func TestKeyServiceWithSet(T *testing.T) {
+	RunWhenKong(T, ">=3.1.0")
+	assert := assert.New(T)
+	require := require.New(T)
+
+	client, err := NewTestClient(nil, nil)
+	assert.NoError(err)
+	assert.NotNil(client)
+
+	createdKeySet, err := client.KeySets.Create(defaultCtx, &KeySet{Name: String("foo")})
+	assert.NoError(err)
+	require.NotNil(createdKeySet)
+
+	//nolint:lll // JSON can't split lines so the encoded field is way too long for Go
+	key := &Key{
+		Name: String("foo"),
+		Kid:  String("foo-1"),
+		Set:  &KeySet{ID: createdKeySet.ID},
+		Jwk: String(`{
+			"kty": "RSA",
+			"kid": "foo-1",
+			"n": "v2KAzzfruqctVHaE9WSCWIg1xAhMwxTIK-i56WNqPtpWBo9AqxcVea8NyVctEjUNq_mix5CklNy3ru7ARh7rBG_LU65fzs4fY_uYalul3QZSnr61Gj-cTUB3Gy4PhA63yXCbYRR3gDy6WR_wfis1MS61j0R_AjgXuVufmmC0F7R9qSWfR8ft0CbQgemEHY3ddKeW7T7fKv1jnRwYAkl5B_xtvxRFIYT-uR9NNftixNpUIW7q8qvOH7D9icXOg4_wIVxTRe5QiRYwEFoUbV1V9bFtu5FLal0vZnLaWwg5tA6enhzBpxJNdrS0v1RcPpyeNP-9r3cUDGmeftwz9v95UQ",
+			"e": "AQAB",
+			"alg": "A256GCM"
+		}`),
+	}
+
+	createdKey, err := client.Keys.Create(defaultCtx, key)
+	assert.NoError(err)
+	require.NotNil(createdKey)
+
+	key, err = client.Keys.Get(defaultCtx, createdKey.ID)
+	assert.NoError(err)
+	require.NotNil(key)
+
+	key.Name = String("bar")
+	key, err = client.Keys.Update(defaultCtx, key)
+	assert.NoError(err)
+	require.NotNil(key)
+	assert.Equal("bar", *key.Name)
+
+	err = client.Keys.Delete(defaultCtx, createdKey.ID)
+	assert.NoError(err)
+
+	// ID can be specified
+	id := uuid.NewString()
+	//nolint:lll // JSON can't split lines so the encoded field is way too long for Go
+	key = &Key{
+		Name: String("foo"),
+		ID:   String(id),
+		Set:  &KeySet{ID: createdKeySet.ID},
+		Kid:  String("foo-2"),
+		Jwk: String(`{
+			"kty": "RSA",
+			"kid": "foo-2",
+			"n": "v2KAzzfruqctVHaE9WSCWIg1xAhMwxTIK-i56WNqPtpWBo9AqxcVea8NyVctEjUNq_mix5CklNy3ru7ARh7rBG_LU65fzs4fY_uYalul3QZSnr61Gj-cTUB3Gy4PhA63yXCbYRR3gDy6WR_wfis1MS61j0R_AjgXuVufmmC0F7R9qSWfR8ft0CbQgemEHY3ddKeW7T7fKv1jnRwYAkl5B_xtvxRFIYT-uR9NNftixNpUIW7q8qvOH7D9icXOg4_wIVxTRe5QiRYwEFoUbV1V9bFtu5FLal0vZnLaWwg5tA6enhzBpxJNdrS0v1RcPpyeNP-9r3cUDGmeftwz9v95UQ",
+			"e": "AQAB",
+			"alg": "A256GCM"
+		}`),
+	}
+
+	createdKey, err = client.Keys.Create(defaultCtx, key)
+	assert.NoError(err)
+	assert.NotNil(createdKey)
+	assert.Equal(id, *createdKey.ID)
+
+	err = client.Keys.Delete(defaultCtx, createdKey.ID)
+	assert.NoError(err)
+
+	err = client.KeySets.Delete(defaultCtx, createdKeySet.ID)
+	assert.NoError(err)
+}
+
+func TestKeyWithTags(T *testing.T) {
+	T.Skip("Key without set fails until Kong/kong@302f2f7")
+	RunWhenKong(T, ">=3.1.0")
+	assert := assert.New(T)
+	require := require.New(T)
+
+	client, err := NewTestClient(nil, nil)
+	assert.NoError(err)
+	assert.NotNil(client)
+
+	//nolint:lll // JSON can't split lines so the encoded field is way too long for Go
+	key := &Key{
+		Name: String("foo"),
+		Kid:  String("foo-1"),
+		Jwk: String(`{
```

Comment:
```suggestion
		KID:  String("foo-1"),
		JWK: String(`{
```

Ditto throughout the rest of the tests; omitting the rest of the suggests because there's a lot of them.